include/base/libmesh.h

```diff
@@ -119,7 +119,7 @@ class LibMeshInit
    * handling.  Finalizes any of Slepc, Petsc, and MPI which were
    * initialized by LibMeshInit.
    */
-  virtual ~LibMeshInit();
+  virtual ~LibMeshInit() noexcept(false);
```

Comment:
I'd rather not be throwing exceptions from the `LibMeshInit` destructor (or any destructor for that matter) since it's generally considered bad practice (destructors are called during stack unwinding, throwing another exception during stack unwinding immediately terminates the program). We use `libmesh_exceptionless_assert()` and `CHKERRABORT` in the `LibMeshInit` destructor for this reason.

Reply:
Agreed.

Reply:
I chose to just revert to using `CHKERRABORT` instead of calling `PetscCallAbort` for compatibility with PETSc < 3.17.0.